src/app/dataset-view/additional-components/metadata-component/components/add-polling-source/edit-polling-source.service.ts

```diff
@@ -0,0 +1,176 @@
+                    return h.history[0].blockHash as string;
+                }),
+                switchMap((hash: string) =>
+                    zip(
+                        this.blockService.onMetadataBlockAsYamlChanges,
+                        this.blockService.requestMetadataBlock(info, hash),
+                    ),
+                ),
+            );
+    }
+
+    private patchFetchStep(
+        sectionForm: FormGroup,
+        editFormValue: EditFormType,
+    ): void {
+        sectionForm.patchValue(editFormValue.fetch);
+        const eventTimeGroup = sectionForm.controls.eventTime as FormGroup;
+        const pattern = editFormValue.fetch.eventTime?.pattern;
+        const timestampFormat = editFormValue.fetch.eventTime?.timestampFormat;
+        eventTimeGroup.addControl("pattern", this.fb.control(pattern));
+        eventTimeGroup.addControl(
+            "timestampFormat",
+            this.fb.control(timestampFormat),
+        );
+        if (editFormValue.fetch.kind === FetchKind.URL) {
+            const headers = sectionForm.controls.headers as FormArray;
+            if ((headers.value as NameValue[]).length == 0)
+                editFormValue.fetch.headers?.forEach((item) => {
+                    headers.push(
+                        this.fb.group({
+                            name: [item.name],
+                            value: [item.value],
+                        }),
+                    );
+                });
+        } else if (editFormValue.fetch.kind === FetchKind.CONTAINER) {
+            const env = sectionForm.controls.env as FormArray;
+            editFormValue.fetch.env?.forEach((item) => {
+                env.push(
+                    this.fb.group({
+                        name: [item.name],
+                        value: [item.value],
+                    }),
+                );
+            });
+            const command = sectionForm.controls.command as FormArray;
+            editFormValue.fetch.command?.forEach((item) => {
+                command.push(this.fb.control(item, RxwebValidators.required()));
+            });
+            const args = sectionForm.controls.args as FormArray;
+            editFormValue.fetch.args?.forEach((item) => {
+                args.push(this.fb.control(item, RxwebValidators.required()));
+            });
+        }
+    }
+
+    private patchReadStep(
+        sectionForm: FormGroup,
+        editFormValue: EditFormType,
+    ): void {
+        sectionForm.patchValue(editFormValue.read);
+        const schema = sectionForm.controls.schema as FormArray;
+        if (
+            !(schema.value as string[]).length &&
+            editFormValue.read.schema &&
+            editFormValue.read.schema.length
+        ) {
+            editFormValue.read.schema.forEach((item) => {
+                schema.push(
+                    this.fb.group({
+                        name: [item.split(" ")[0]],
+                        type: [item.split(" ")[1]],
```

Comment:
I guess splitting item once should be enough

Reply:
Refactored